XIPs/xip-10-brandinfo-schema.md

```diff
@@ -0,0 +1,90 @@
+---
+xip: 14
+title: Conversation context metadta schema
+description: Schema to enable standardized sharing of conversation context
+author: Yash Lunagaria (@yash-luna)
+status: Review
+type: Standards Track
+category: XRC
+created: 2023-01-31
+---
+
+## Abstract
+
+This XIP proposes a schema for sharing additional context about a conversation beyond a `conversationId`. Having a standard way to provide not only a conversationId, but also conversation metadata will enable different frontends to display useful contextual information in a consistent, user-friendly manner, while still affording apps design and UI flexibility.
+
+## Motivation
+
+The SDK API currently accepts a unique `conversationId` per address pair and allows `metadata` to be set as well. Today, different apps use these parameters in different ways. Therefore an app may not know how to correctly render in its UI `conversationId` and `metadata` for conversations originating from other apps. We propose a schema to standardize the use of conversation `metadata` for the purpose of app brand expression stemming from the `conversationId`. The `metadata` field may include other non-standard (or future standard) properties.
+
+Note that the schema is only applicable when setting a non-null `conversationId`. If `conversationId` is null, the SDK does not allow setting `metadata`.
+
+## Specification
+
+Proposed `displayInfo` schema in conversation `metadata`
+
+```json
+{
+    conversationId: "mydomain.xyz/abc/qrs",
+    metadata: { 
+        "displayInfo.prettyName": "My company", 
+        "displayInfo.profileImage": "mydomain.xyz/assets/myimage.png", 
+        "displayInfo.primaryColor": "#ffffff" }
+}
+```
+
+Example `displayInfo` implementation for a chat app named Galaxy
+
+```json
+{
+    conversationId: "galaxy.chat/dm/uniqueIdentifier",
+    metadata: { 
+        "displayInfo.prettyName": "Galaxy", 
+        "displayInfo.profileImage": "galaxychat.xyz/brandassets/logo.png", 
+        "displayInfo.primaryColor": "#6865B8" }
+}
+```
+
+The `profileImage` must meet the following criteria:
+
+- Aspect ratio: 1:1
+- Minimum resolution: 100x100
+- Maximum resolution: 800x800
+- Format: .PNG, .WEBP
+
+The `primaryColor` must be a hex color code.
```

Comment:
How is a fallback color different from a second hex code? Didn't understand how 'a fallback color would allow the front-end to set up their own interpretation of the colors'

Reply:
@yash-luna to use an analogy, a "fallback" color in this case is the description of a color of paint, whereas the hex code would be similar to the paint code instructions on the outside of a can of paint.

In the fallback case the specific color used would be based off of the developer's interpretation. With the hex color there is no interpretation and it is explicit.

The naming scheme for the fallback could be based off of an existing example, such as Material Design or Tailwind CSS's color palette. It usually has a color name and separate value to indicate darkness or saturation.

If given the choice, if I were developing an app I would always pick the fallback color as I'd have a bit more control over how it should be represented. Without that control, I might ignore the color altogether as the results could really clash with my experience.